### PR TITLE
#16: implement new user view model data per git hub issue

### PR DESCRIPTION
**Pull request by ThreedTX**
Made changes necessary to create a dropdown of a user's collections in the spots view

### CODE REVIEW

File: routes/spots.js [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
 
-router.get('/:id', async (req, res, next) => {
-    const id = req.params.id;
+router.get("/:id", async (req, res, next) => {
+  const id = req.params.id;
+  const userId = req.session.auth.userId;
 
-    const spot = await Spot.findByPk(id);
-    const reviews = await Review.findAll({where: { spotId: id}});
+  const spot = await Spot.findByPk(id);
```
Comment (BenBolthouse) [inline routes/spots.js:15]: Combine these two queries using `include` statement for the reviews.

File: routes/spots.js [hunk 2]
```diff
@@ -1,19 +1,44 @@
...
+  const reviews = await Review.findAll({ where: { spotId: id } });
+  const customCollections = await Collection.findAll({
```
Comment (BenBolthouse) [inline routes/spots.js:17]: Query to find all collections for the user, but then use javascript array methods to extract the two default collections to a separate array.

File: routes/spots.js [hunk 3]
```diff
@@ -1,19 +1,44 @@
...
+  res.render("spot", {
+    title: `${spot.name}`,
+    spot,
+    reviews,
+    user: {
+      firstName,
+      lastName,
+      username,
+      email,
+      customCollections: [customCollections],
```
Comment (BenBolthouse) [inline routes/spots.js:38]: I believe spread operator is needed here, please test...thanks!

**Review by BenBolthouse — COMMENTED** [review]
Looks good!!

File: routes/spots.js [hunk 4]
```diff
@@ -1,19 +1,44 @@
...
+      userId: userId,
+      name: { [Op.or]: ["Want To Visit", "Have Visited"] },
+    },
+  });
+  const { firstName, lastName, username, email } = req.session.user;
 
-    res.render('spot', { title: `${spot.name}`, spot, reviews, user: req.session.user });
-})
+  res.render("spot", {
+    title: `${spot.name}`,
```
Comment (BenBolthouse) [inline routes/spots.js:30]: `title` object is redundant with `spot` object in scope.

**Review by BenBolthouse — CHANGES_REQUESTED** [review]

File: routes/spots.js (hunk 2 above)
Comment (Austin-from-TX) [reply routes/spots.js:21]: Mitchell helped me come up with these 2 queries that should grab all the Reviews, Collections and SpotCollections tied to our current user:  spot variable to be used to render the spot info with associated reviews.  user variable to be used to perform client side validation checks to determine how to render our buttons on page load && avoid duplicate entries to our DB.  hopefully this helps, let me know if further clarification is needed. 

    const id = req.params.id;
    const userId = req.session.auth.userId;
    
    const spot = await Spot.findOne({where: {id}, include: Review });

    const user = await User.findOne({ where: { id: userId }, include: { model: Collection, as: 'collections', include: { model: Spot,   where: { id } } } } );


//use res.json(spot/user) and open page in browser to view expected output for each

File: routes/spots.js [hunk 5]
```diff
@@ -1,19 +1,57 @@
...
+  const spot = await Spot.findByPk(id);
+  // const reviews = await Review.findAll({ where: { spotId: id } });
 
-    res.render('spot', { title: `${spot.name}`, spot, reviews, user: req.session.user });
-})
+  // Optimize the above query to do eager load of Review
+
+
+  const customCollections = await Collection.findAll({
+    where: { userId: userId },
```
Comment (BenBolthouse) [inline routes/spots.js:22]: Heads up: I ran this code without being an authenticated user and it threw an error. Make sure to you build a check into your logic to determine if the user is signed in. If they aren't then don't send the collections in the response. To replicate the error try going to a spot landing page such as `http://localhost:8080/spots/6` while browsing the site unauthenticated. I.e., `req.session.auth.userId` is undefined if the user is not logged in.

**Review by genagain — COMMENTED** [review]
Looks good to me!

**Pull request merged**